### PR TITLE
AcctIdx: only advance age on thread 0

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem

Remove potential race condition by limiting age advancement to 1 special thread in the flushing pool.

#### Summary of Changes

Fixes #

### CODE REVIEW

**Review by brooksprumo — COMMENTED** [review]
lgtm. Do you think there's benefit in a test where `can_advance_age` is set to `false` in one of the threads/fn calls?

**Review by brooksprumo — APPROVED** [review]

**Comment by jeffwashington** [comment]
> lgtm. Do you think there's benefit in a test where `can_advance_age` is set to `false` in one of the threads/fn calls?

I do not understand. Can you please el
aborate?

**Comment by brooksprumo** [comment]
> > lgtm. Do you think there's benefit in a test where `can_advance_age` is set to `false` in one of the threads/fn calls?
> 
> I do not understand. Can you please elaborate?

I was thinking of a test case where `background()` is called from a thread (or threads) where initially `can_advance_age` is set to false. The test would ensure that all buckets are flushed at the current age, and that the age does not advance. Then call `background()` from a new thread or call a fn that can advance the age and check again.

File: runtime/src/accounts_index_storage.rs [hunk 1]
```diff
@@ -71,7 +74,7 @@ impl BgThreads {
                     Builder::new()
                         .name("solana-idx-flusher".to_string())
```
Comment (HaoranYi) [inline runtime/src/accounts_index_storage.rs:75]: can we add thread `idx` to thread name? might be useful for future debugging.

**Review by HaoranYi — APPROVED** [review]
lgtm

**Pull request merged**